addons/sale/models/payment_token.py

```diff
@@ -0,0 +1,24 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class PaymentToken(models.Model):
+    _inherit = 'payment.token'
+
+    def _get_available_tokens(self, order_sudo, providers_sudo, logged_in, **kwargs):
+        """ Override of `payment` to return the tokens of the sale order' available providers.
+
+        :param `sale.order` order_sudo: The sales order being paid.
+        :param `payment.provider` providers_sudo: The providers available for the transaction.
```

Comment:
The method would be more convenient to use (in future calls?) if it handled provider ids instead of a sudoed recordset.